plugins/woocommerce/client/legacy/js/frontend/checkout.js

```diff
@@ -954,19 +956,38 @@ jQuery( function ( $ ) {
 				text: msg,
 			} ).appendTo( $target );
 		},
-		remove_coupon_error: function ( evt ) {
-			$( evt.currentTarget )
+		remove_coupon_error: function () {
+			var $coupon_field = $( '#coupon_code' );
+
+			if ( $coupon_field.length === 0 ) {
+				return;
+			}
+
+			$coupon_field
 				.removeClass( 'has-error' )
 				.removeAttr( 'aria-invalid' )
 				.removeAttr( 'aria-describedby' )
 				.next( '.coupon-error-notice' )
 				.remove();
 		},
+
+		clear_coupon_input: function () {
+			const $coupon_field = $( '#coupon_code' );
+			$coupon_field
+				.val('')
+				.removeClass('has-error')
+				.removeAttr('aria-invalid')
+				.removeAttr('aria-describedby')
+				.next('.coupon-error-notice')
+				.remove();
+		},
 		submit: function ( evt ) {
 			var $form = $( evt.currentTarget );
 			var $coupon_field = $form.find( '#coupon_code' );
 			var self = this;
 
+      self.remove_coupon_error();
```

Comment:
Please can we keep indentation consistent, we have added spaces here but the rest of the file uses tabs.